go/ethadapter/geth_rpc_client.go

```diff
@@ -55,11 +55,7 @@ func (e *gethRPCClient) FetchHeadBlock() (*types.Block, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), e.timeout)
 	defer cancel()
 
-	blk, err := e.client.BlockByNumber(ctx, nil)
-	if err != nil {
-		e.logger.Crit("could not fetch head block.", log.ErrKey, err)
-	}
-	return blk, nil
+	return e.client.BlockByNumber(ctx, nil)
```

Comment:
is there never a possibility of an error that doesn't get fixed by retrying?

Reply:
There might be, but I don't think the client should be fatally shutting down the process, it was a bit hacky having the Crit in here I think (my bad).

The caller of the client can look at the error and decide how to handle it (and I've just added a note to the infinite retry strategy to remind users that they can use `retry.FailFast(err)` wrapper to break out of the retry if needed).